VERSIONING.md

```diff
@@ -126,6 +126,12 @@ is designed so the following goals can be achieved.
       stable release except for a stable release of the contrib repository.
 * GitHub releases will be made for all releases.
 * Go modules will be made available at Go package mirrors.
+* This repository's modules are only guaranteed to be compatible with
+  modules released from the same commit. For example,
```

Comment:
It should be clear here that we are talking cross module compatibility by using the term "other modules" and we want to guarantee across the release not a commit.

```suggestion
* Go modules are only guaranteed to be compatible with
  other modules from the same release. They are not
  guaranteed to be compatible with other modules from
  previous releases.  For example,
```